modules/nf-core/cafe/tests/main.nf.test

```diff
@@ -0,0 +1,58 @@
+nextflow_process {
+
+    name "Test Process CAFE"
+    script "../main.nf"
+    process "CAFE"
+
+    tag "modules"
+    tag "modules_nfcore"
+    tag "cafe"
+
+    test("mycoplasma - genecounts - tree") {
+        when {
+            process {
+                """
+                input[0] = channel.of( file('s3://excon/cafe_input/hog_gene_counts.tsv') )
+                input[1] = channel.of( file('s3://excon/cafe_input/SpeciesTree_rooted_ultra.txt') )
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert process.out.cafe_base_count != null },
+                { assert snapshot(
+                    process.out.cafe_base_count
+                    ).match()
```

Comment:
This will check that most of the channels are stable, except the `cafe_report` where I picked some lines to match.
It wasn't obvious why that wasn't stable but I assumed it was probably machine precision differences.
Note that if you add a meta map then you'd need `file(process.out.cafe_report[0][1])` to access the second part.

Reply:
Great, ok, thats a much better snapshot